### DISCUSSION TITLE
[Deployment] Front End on Vercel, Back End/Database on Render?

### DISCUSSION THREAD

**Discussion by matepapp** [blitz-js/blitz#528] [General] [answered]
Hey! 👋 

Awesome work on Blitz.js so far! I'm starting to investigate whether it fits my next project requirements. 

Does it make sense to deploy the Front End and API routes on Vercel (since it's basically a Next.js app that runs the best on Vercel as far as I know) and the Back End / Database on a managed hosted solution such as Render? Or what is the currently recommended deployment strategy? I've read that you're checking the [Serverless Database connection metrics](https://github.com/blitz-js/blitz/discussions/503).

**Comment by Zeko369** [comment, marked as answer]
IMO, currently you should just stick to render/DO for both server and DB on the same machine or you can still split them.

1. Prisma currently has a bug in serverless mode with not closing connections property

2. You can get a lot better performance running on a traditional server

3. Heroku is free, DO has 15usd for single server where you can host both, compared to Vercel where you need to pay if you have a lot of models  (they have a limit of 12 functions in free I think, we're planning to do function API function bundling but as things stand 12 is the free limit)

Once there is a way to [change pgBouncer settings on DO](https://github.com/prisma/prisma/issues/1638#issuecomment-628235096) we'll be able to greatly improve performance on Vercel, also we're planning on bundling same level api functions (for example whole CRUD backend would be one function, reducing it from 5 to 1) which solved the running out of free functions problem.

Also I head good things about [Vultr](vultr.com), haven't tried them myself, so maybe try that if you're interested in self hosting. or run the whole thing on render (there is a render.com tutorial in USER_GUIDE, but seems to not be working for some reason)

**Reply by ryardley** [reply]
For the record DO have had a fix for pgBouncer for some time. Blitz works fine on serverless.

**Comment by Zeko369** [comment]
Also you mentioned FrontEnd on Vercel and BackEnd/DB on Render, there is not way to do exactly that since the whole app is fullstack which tightly integrated backend and frontend into a next app (with API routes) so it would be quite hard to deploy them separately

**Comment by matepapp** [comment]
Thank you for the explanation @Zeko369! In this case, I think I'd stick with a traditional server on Render.
